mono/metadata/icall.c

```diff
@@ -766,13 +785,15 @@ ves_icall_System_Array_GetLowerBound (MonoArray *arr, gint32 dimension)
 }
 
 ICALL_EXPORT void
-ves_icall_System_Array_ClearInternal (MonoArray *arr, int idx, int length)
+ves_icall_System_Array_ClearInternal (MonoArrayHandle arr, int idx, int length, MonoError* error)
```

Comment:
Use `gint32` instead of `int`: the former is guaranteed to be the same as managed `int` while the former isn't.